2_ThreeSum.cpp

```diff
@@ -62,31 +62,31 @@ using namespace std;
 vector<vector<int> > threeSum(vector<int> &num)
 {
 	//sort the array
-	sort(num.begin(), num.end());
+    sort(num.begin(), num.end());
     //ret contains the resultant triplets
     vector<vector<int> > res;
     //tuple format of the result
     vector<int> tuples(3);
     for (int i = 0; i < num.size(); ++i) 
-	{
+    {
       // ignore duplicate elements
     	if (i > 0 && num[i] == num[i-1])
     		continue;
       	int target = -num[i];
       	int start = i+1, end = num.size()-1;
       	while (start < end) 
-		{
+	{
 			//found a triplet with sum as 0
         	if (num[start] + num[end] == target) 
-			{
+		{
          		tuples[0] = num[i];
           		tuples[1] = num[start];
           		tuples[2] = num[end];
           		res.push_back(tuples);
           		++start;
           		--end;
-        	} 
-			else if (num[start] + num[end] > target)
+        	} 	
+		else if (num[start] + num[end] > target)
           		--end;
         	else 
           		++start;  

```

Comment:
line 97: if res is empty, can't you just return it rather than a new vector<vector<int>>() as they would be equivalent?

Reply:
it isn't letting me comment below this point, so I'll put those comments here

Reply:
I'm not familiar with how the sort function works, will it truly sort a vector of vectors as expected? I'm not sure that it would.  Also, can you think of optimizations to the above code that would make this step unnecessary?

Reply:
Yes, the sort functions would return a sorted vector of vectors. I wasn't able to think of any optimizations further as I'm using Binary search in my algorithm and that requires the elements to be a sorted first. Will still think of anything I can do to avoid this step.